File: src/test/java/com/flowcommerce/core/domain/member/MemberTest.java
package com.flowcommerce.core.domain.member;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class MemberTest {

    @Test
    void 일반_회원가입이_가능하다() {
        // given
        String email = "test@test.net";
        String password = "Password!";
        String name = "first";

        // when
        Member member = Member.create(name, email, password);

        // then
        assertThat(member.getEmail()).isEqualTo(email);
        assertThat(member.getName()).isEqualTo(name);
        assertThat(member.getAuthProvider()).isEqualTo(AuthProvider.LOCAL);
        assertThat(member.getStatus()).isEqualTo(MemberStatus.ACTIVE);
        assertThat(member.getProviderId()).isNull();
    }

}

File: src/test/kotlin/com/flowcommerce/core/domain/member/MemberTest.kt
package com.flowcommerce.core.domain.member

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MemberTest {

    @Test
    fun `일반 회원가입이 가능하다`() {
        // given
        val email = "test@test.net"
        val password = "Password!"
        val name = "first"

        // when
        val member = Member.create(name, email, password)

        // then
        assertThat(member.email).isEqualTo(email)
        assertThat(member.name).isEqualTo(name)
        assertThat(member.authProvider).isEqualTo(AuthProvider.LOCAL)
        assertThat(member.status).isEqualTo(MemberStatus.ACTIVE)
        assertThat(member.providerId).isNull()
    }
}
